test/authentication_middleware_test.rb

```diff
@@ -2,7 +2,7 @@
 
 class AuthenticationMiddlewareTest < Faraday::TestCase
   def conn
-    Faraday::Connection.new('http://example.net/') do |builder|
+    Faraday::Connection.new do |builder|
```

Comment:
Why did you have to remove this?

Reply:
Specifying a domain in the connection forces the stubs to only be available for that domain. Therefore, making `connection.get('/path')` (without the host) won't match. I removed this to make the connection domainless.

Reply:
See my new comment above on the expected behaviour.
I hope this is clear now, this test should definitely work with the domain in place (it will test point 2 of my comment).

Rather than changing existing tests, your approach should be to duplicate them in order to test both points (old and new behaviour).